tests/ShadowDOM/js/test.js

```diff
@@ -520,4 +524,40 @@ suite('Shadow DOM', function() {
 
   });
 
+  // https://github.com/webcomponents/webcomponentsjs/issues/316
+  test('nested renderer should invalidate attributes', function() {
+    var div = document.createElement('div');
+    div.innerHTML =
+        '<x-outer></x-outer>' +
+        '<template id="x-host">' +
+          'Foo: [<content select="[foo]"></content>] ' +
+          'Bar: [<content select="[bar]"></content>]' +
+        '</template>' +
+        '<template id="x-outer">' +
+          '<x-host id="host">' +
+            '<child foo>~Foo~</child> ' +
+            '<child foo bar>~Foo+Bar~</child> ' +
```

Comment:
if the trailing space is removed here, things go really haywire, even reverting the ShadowRenderer fix. SD polyfill might be expecting text nodes between distributed elements (and they almost always are there in practice). Need to investigate more, but probably best to keep separate from this fix.